File: xide-ui/src/main/java/org/xast/xide/ui/dialog/SettingsDialog.java
package org.xast.xide.ui.dialog;

import java.awt.BorderLayout;

import javax.swing.JDialog;
import javax.swing.JLabel;

import org.xast.xide.ui.MainFrame;

public class SettingsDialog extends JDialog {
    public SettingsDialog(MainFrame owner) {
        super(owner.getFrame(), "Settings", true); // modal

        setSize(700, 500);
        setLocationRelativeTo(owner.getFrame());
        setLayout(new BorderLayout());

        add(new JLabel("Settings go here"), BorderLayout.CENTER);
    }
}

File: xide-standard-plugins/xide-settings-plugin/src/main/kotlin/org/xast/xide/settings_plugin/SettingsDialog.kt
package org.xast.xide.settings_plugin

import java.awt.BorderLayout
import javax.swing.JDialog
import javax.swing.JFrame
import javax.swing.JLabel

class SettingsDialog(owner: JFrame) : JDialog(owner, "Settings", true) {
    init {
        setSize(700, 500)
        setLocationRelativeTo(owner)
        layout = BorderLayout()

        add(JLabel("Settings go here"), BorderLayout.CENTER)
    }
}